### ISSUE TITLE
[expr~] Changing the array index won't work

### ISSUE THREAD

**Issue by gabochi** [pure-data/pure-data#719]
It is possible to change the array index on [expr] depending on the input. For example:

expr array1[$f1%10]

But, when working with signals, it doesn't work (just fixed index will be recognized):

expr~ array[10] ..........is OK
expr~ array1[$v1] ......doesn't work

It is truth that you could replace this expression with a [tabread~] but I would like to implement it in more complex expressions, combining several operations, as expected with expr~.

**Comment by porres** [comment]
you need to use floats, $v# will not work for array indexes

**Comment by porres** [comment]
$f# will work, or you can also use a variable from [value]

**Comment by porres** [comment]
check the help file of [expr], I've put examples there in [pd Arrays].

**Comment by porres** [comment]
unless you want to request that it'd take signals instead, which is fine. I don't know what's preventing it

**Comment by Spacechild1** [comment]
> unless you want to request that it'd take signals instead,

I think that's what he's saying ;-)

**Comment by porres** [comment]
yeah, I didn't read it carefully, I see now he's mentioning how one can use '$f1', sorry...

Can I propose the title change to:

request: allow signals to set array's index in 'expr~'

**Comment by gabochi** [comment]
Yes, totally, that's what I meant (and the reason I put **$v1** in the example). Do you think it is possible and keep it recursive still like in **expr~ array[$v1/1000%3+5]** or so?

Thank you all.

**Comment by porres** [comment]
I don't understand what you mean by "recursive still", but it surely seems possible to add this feature

**Comment by gabochi** [comment]
Yes, doesn't matter, just allow signals ($v?) to set array index in **expr~** like you said.

**Comment by porres** [comment]
are you planning to do some weird rampcode stuff? :)

**Comment by gabochi** [comment]
I cannot hide it anymore...

**Comment by nivlekp** [comment]
An attempted solution is up. #853

**Comment by porres** [comment]
For the record, you can use signals to read arrays with [fexpr~]
